### PR TITLE
Allow to install caddy from repos

### LINKED ISSUE #62: Add support for .deb / .rpm repositories
Beginning with Caddy 2, there will be official packages available:
* https://caddyserver.com/docs/install#official-packages

### PR DESCRIPTION
**Pull request by jay7x**
Closes #62

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -88,17 +91,29 @@
...
+# @param package_name
+#   Name of the caddy package to use. Only relevant when $install_method is 'repo'.
+#
+# @param package_ensure
+#   Whether to install or remove the caddy package. Only relevant when $install_method is 'repo'.
 #
 class caddy (
   String[1]                      $version                         = '2.0.0',
-  Optional[Enum['github']]       $install_method                  = undef,
+  Optional[Enum['github','repo']] $install_method                 = undef,
```
Comment (smortex) [inline manifests/init.pp:116]: I am wondering if we should remove the 'Optional' part and add something like 'upstream'. Some people use Hiera to change settings in modules, and the current way make it impossible to revert to upstream on one host if another level in the hierarchy set it to something else.

Comment (jay7x) [reply manifests/init.pp:116]: Yes, I have that in my mind, but it'll be breaking change (potentially, if someone do `class { 'caddy': install_method => undef }`). So far I'm trying to keep backward compatibility.

Comment (jay7x) [reply manifests/init.pp:116]: I mean, I'll change this, just in different PR.

**Comment by jay7x** [comment]
https://github.com/voxpupuli/puppet-caddy/pull/115 is required to be merged before to fix acceptance tests.

**Comment by jay7x** [comment]
Debian family is good now. Need to fix puppet-yum to make RedHat family happy.

**Review by bastelfreak — APPROVED** [review]

**Pull request merged**